extensions/2.0/Khronos/KHR_materials_iridescence/README.md

```diff
@@ -0,0 +1,207 @@
+See [Appendix](#appendix-full-khronos-copyright-statement) for full Khronos Copyright Statement.
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Exclusions
+
+* This extension must not be used on a material that also uses `KHR_materials_pbrSpecularGlossiness`.
+* This extension must not be used on a material that also uses `KHR_materials_unlit`.
+
+## Overview
+
+Iridescence describes an effect where hue varies depending on the viewing angle and illumination angle: A thin-film of a semi-transparent layer results in inter-reflections and due to thin-film interference, certain wavelengths get absorbed or amplified. Iridescence can be seen on soap bubbles, oil films, or on the wings of many insects.
+With this extension, thickness and index of refraction (IOR) of the thin-film can be specified, enabling iridescent materials. 
+
+## Extending Materials
+
+The iridescence materials are defined by adding the `KHR_materials_iridescence` extension to any glTF material.
+
+```json
+{
+    "materials": [
+        {
+            "extensions": {
+                "KHR_materials_iridescence": {
+                    "iridescenceFactor": 1.0,
+                    "iridescenceIOR": 1.8,
+                    "iridescenceThicknessMaximum": 1200.0
+                }
+            }
+        }
+    ]
+}
+```
+
+### Iridescence
+
+All implementations should use the same calculations for the BRDF inputs. Implementations of the BRDF itself can vary based on device performance and resource constraints. See [appendix](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#appendix-b-brdf-implementation) for more details on the BRDF calculations.
+
+|                              | Type                                                                | Description                                                                                      | Required              |
+|------------------------------|---------------------------------------------------------------------|--------------------------------------------------------------------------------------------------|-----------------------|
+|**iridescenceFactor**         | `number`                                                            | The iridescence intensity.                                                                       | No, default: `0.0`    |
+|**iridescenceTexture**        | [`textureInfo`](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#reference-textureinfo) | The iridescence intensity stored in the red channel of the texture.                                                               | No                    |
+|**iridescenceIOR**               | `number`                                                            | The index of refraction of the dielectric thin-film layer.                                       | No, default: `1.8`    |
```

Comment:
Why is `iridescenceIOR` not texturable? In contrast to KHR_materials_ior which defines the IOR of the 3D volume, the thin film IOR is a 2D surface effect, so texturing is easily possible. I understand that due to several technical reasons less textures is better than more textures, but it feels somewhat arbitrary. What if a texture is used as an atlas to represent different materials on a single shape? Maybe the reason for this decision should be stated somewhere in the spec.

Reply:
As discussed on Monday's call, we decided that IOR does not need to be texturable. Our opinion is that due to the physical process of how the film is applied via spray, the IOR is unlikely to vary across a material and does not warrant a dedicated texture.